src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs

```diff
@@ -99,7 +99,20 @@ protected override LineColumnRule GetLineColumnRuleBetween(SyntaxTrivia trivia1,
 
                 if (trivia2.IsKind(SyntaxKind.RegionDirectiveTrivia) || trivia2.IsKind(SyntaxKind.EndRegionDirectiveTrivia))
                 {
-                    return LineColumnRule.PreserveLinesWithDefaultIndentation(lines);
+                    var previous = trivia2;
+                    while ((previous = previous.GetPreviousTrivia(previous.SyntaxTree, CancellationToken.None)).IsKind(SyntaxKind.WhitespaceTrivia))
+                    {
+                        // skip whitespace trivia.
```

Comment:
needs more explanation :)

Reply:
what about end-of-line trivia?

Reply:
> what about end-of-line trivia?

The following test just passes:

```csharp
        [Fact]
        public async Task TestRegion2()
        {
            var testCode = @"
class MyClass
{
#if true
    public void M()
    {
[||]#region ABC
        System.Console.WriteLine();
[||]#endregion
    }
#else
    public void M()
    {
#region ABC
        System.Console.WriteLine();
#endregion
    }
#endif
}
";
            var fixedCode = @"
class MyClass
{
#if true
    public void M()
    {
        #region ABC
        System.Console.WriteLine();
        #endregion
    }
#else
    public void M()
    {
#region ABC
        System.Console.WriteLine();
#endregion
    }
#endif
}
";
            await Verify.VerifyCodeFixAsync(testCode, fixedCode);
        }
```

(I'll definitely commit and push it)

`'#region'.GetPreviousTrivia(..)` is not an end of line trivia for this case though.

Do you have an interesting test case where the tree will have end of line trivia?

I found a niche case where my logic is incorrect apart from EOL trivia:

```csharp
#region OUTER
        #region ABC
        System.Console.WriteLine();
        #endregion
#endregion
```

Here, after we go back consuming whitespace trivia, we don't find a disabled text, but we are in disabled text.

Is there a compiler API for the "Is this position conditionally excluded?" question?

Reply:
does IsInInactiveRegion work?

Reply:
@CyrusNajmabadi It doesn't 😕 

https://github.com/dotnet/roslyn/blob/75bbc51a05b71d8a07c861201660f55974909845/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTreeExtensions.cs#L477-L538

That seems a hard-to-answer question on the IDE side.

Reply:
> That seems a hard-to-answer question on the IDE side.

Not really. I think I see how I can do it.

Reply:
@CyrusNajmabadi This is ready for another look now :)